Content.Server/Weather/WeatherSystem.cs

```diff
@@ -0,0 +1,91 @@
+using System.Linq;
+using Content.Server.Administration;
+using Content.Shared.Administration;
+using Content.Shared.Weather;
+using Robust.Shared.Console;
+using Robust.Shared.GameStates;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Weather;
+
+public sealed class WeatherSystem : SharedWeatherSystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<WeatherComponent, ComponentGetState>(OnWeatherGetState);
+    }
+
+    private void OnWeatherGetState(EntityUid uid, WeatherComponent component, ref ComponentGetState args)
+    {
+        args.State = new WeatherComponentState()
+        {
+            Weather = component.Weather,
+            EndTime = component.EndTime,
+            StartTime = component.StartTime,
+        };
+    }
+}
+
+[AdminCommand(AdminFlags.Fun)]
+public sealed class WeatherCommand : IConsoleCommand
+{
+    // Wouldn't you like to know, weather boy.
+    [Dependency] private readonly IEntityManager _entManager = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly IPrototypeManager _protoManager = default!;
+
```

Comment:
can you add this with the `IConsoleHost.RegisterCommand` in the system instead since its already here